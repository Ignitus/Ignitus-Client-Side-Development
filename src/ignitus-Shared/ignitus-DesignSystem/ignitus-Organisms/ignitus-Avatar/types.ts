export type avatar = {
  name: string;
  src: string;
};

/** As AvatarImage represents img feel free to add more props here, as <img /> supports lot of other attributes. */
export type AvatarImageProps = {
  src: string;
  alt: string;
};
