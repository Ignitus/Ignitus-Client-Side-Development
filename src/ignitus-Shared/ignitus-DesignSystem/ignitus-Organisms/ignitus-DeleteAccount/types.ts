export interface PropTypes {
  username: string;
}
