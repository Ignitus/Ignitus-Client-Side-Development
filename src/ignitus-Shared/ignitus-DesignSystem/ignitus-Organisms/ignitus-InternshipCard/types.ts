export type CardTypes = {
  ImageLink: string;
  Title: string;
  Position?: string;
  Place?: string;
  Company?: string;
  Description?: string;
  type?: string;
  JobType?: string;
};

export type CardProps = {
  ImageLink: string;
  Title: string;
  Position?: string;
  Place?: string;
  Company?: string;
  Description?: string;
  type: string;
  JobType?: string;
};

export type StyleProps = {
  size?: string;
};
