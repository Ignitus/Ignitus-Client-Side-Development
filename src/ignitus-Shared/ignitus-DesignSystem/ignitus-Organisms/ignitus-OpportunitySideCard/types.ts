export type OpportunityType = {
  title: string;
  description: string;
};

export type CardType = {
  location: string;
  commitment: string;
  salary: string;
};

export type StatusType = {
  status: string;
};
