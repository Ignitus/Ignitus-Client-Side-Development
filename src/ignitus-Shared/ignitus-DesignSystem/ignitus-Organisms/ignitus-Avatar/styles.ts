import styled from '@emotion/styled';

export const AvatarImage = styled.img`
  max-width: 9rem;
`;
