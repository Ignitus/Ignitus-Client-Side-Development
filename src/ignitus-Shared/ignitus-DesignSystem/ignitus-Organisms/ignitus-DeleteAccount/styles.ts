import styled from '@emotion/styled';
import * as C from '../../ignitus-Atoms/colors';
import { Paragraph } from '../../ignitus-Atoms/typography';
import * as F from '../../ignitus-Atoms/fonts';
import { flexibleRowDiv } from '../../shared';
import { Icon } from '../../../ignitus-Utilities/Components/icon';
import { DefaultCheckbox } from '../../..';
import { RoundedButton } from '../../ignitus-Atoms/buttons';

export const DeleteContainer = styled.div`
  background-color: ${C.White};
  background-size: contain;
  border-radius: 1rem;
  padding: 2rem;
  display: inline-flex;
  flex-direction: column;
  box-shadow: 0 2px 4px 0 ${C.boxShadowColor};
  width: 40rem;
`;

export const TopSection = styled(flexibleRowDiv)`
  justify-content: space-between;
`;

export const MiddleSection = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
`;

export const BottomSection = styled(flexibleRowDiv)`
  justify-content: flex-end;
  margin-top: 2rem;
`;

export const StyledIcon = styled(Icon)`
  width: 1.4rem;
  fill: ${props => props.color};
  transition: 0.3s;
  flex: 1;
  margin-top: 0.2rem;
`;

export const IconContainer = styled.button`
  border: none;
  cursor: pointer;
  overflow: hidden;
  margin: 0 0.2rem;
  height: 2.8rem;
  width: 2.8rem;
  border-radius: 50%;
  background-color: ${props => props.color};
  outline-color: transparent;
`;

export const StyledParagraph = styled(Paragraph)`
  padding: 1rem 0;
`;

export const StyledCheckbox = styled(DefaultCheckbox)`
  font-size: ${F.SM};
  margin-bottom: 1rem;
`;

export const StyledTextArea = styled.textarea`
  background-color: ${C.GreyLight};
  padding: 1rem;
  color: ${C.GreySecondaryText};
  font-weight: ${F.SemiBold};
  font-size: ${F.MD};
  border: none;
  border-radius: 1rem;

  ::placeholder {
    color: ${C.GreySecondaryText};
    font-weight: ${F.SemiBold};
    font-size: ${F.MD};
    opacity: 0.5;
  }
`;

export const StyledRoundedButton = styled(RoundedButton)`
  font-weight: ${F.Bold};
  box-shadow: 0 6px 12px ${C.boxShadowColorLight};
`;
