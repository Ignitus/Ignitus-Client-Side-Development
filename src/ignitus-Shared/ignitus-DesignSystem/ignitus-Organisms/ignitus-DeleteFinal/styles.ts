import styled from '@emotion/styled';
import * as C from '../../ignitus-Atoms/colors';
import { Paragraph } from '../../ignitus-Atoms/typography';
import * as F from '../../ignitus-Atoms/fonts';
import { flexibleRowDiv } from '../../shared';
import { Icon } from '../../../ignitus-Utilities/Components/icon';
import { RoundedButton } from '../../ignitus-Atoms/buttons';

export const DeleteContainer = styled.div`
  background-color: ${C.White};
  background-size: contain;
  border-radius: 1rem;
  display: inline-flex;
  flex-direction: column;
  box-shadow: 0 2px 4px 0 ${C.boxShadowColor};
  width: 45rem;
  padding: 2rem;
`;

export const TopSection = styled(flexibleRowDiv)`
  justify-content: space-between;
`;

export const MiddleSection = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
`;

export const BottomSection = styled(flexibleRowDiv)`
  margin-top: 1rem;
  justify-content: flex-end;
`;

export const StyledIcon = styled(Icon)`
  width: 1.4rem;
  fill: ${props => props.color};
  transition: 0.3s;
  flex: 1;
  margin-top: 0.2rem;
`;

export const IconContainer = styled.button`
  border: none;
  cursor: pointer;
  overflow: hidden;
  margin: 0 0.2rem;
  height: 2.8rem;
  width: 2.8rem;
  border-radius: 50%;
  background-color: ${props => props.color};
  outline-color: transparent;
`;

export const StyledParagraph = styled(Paragraph)`
  padding: 0.5rem 0;
`;

export const StyledText = styled(Paragraph)`
  font-weight: ${F.Bold};
  padding: 0.5rem 0;
`;

export const StyledRoundedButton = styled(RoundedButton)`
  box-shadow: 0 6px 12px ${C.boxShadowColorLight};
`;
