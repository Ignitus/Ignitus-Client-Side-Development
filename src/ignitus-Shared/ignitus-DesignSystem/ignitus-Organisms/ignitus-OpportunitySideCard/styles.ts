import styled from '@emotion/styled';
import { White, GreyBackground, IgnitusBlue } from '../../ignitus-Atoms/colors';
import { SemiBold } from '../../ignitus-Atoms/fonts';
import { Heading4 } from '../../ignitus-Atoms/typography';
import { RoundedButton } from '../../ignitus-Atoms/buttons';
import { Icon as I } from '../../../ignitus-Utilities/Components/icon';
import { StatusType } from './types';

export const Container = styled.div`
  border-radius: 1rem;
  background-color: ${White};
  display: flex;
  flex-direction: column;
`;

export const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  min-width: 18rem;
  padding: 2rem;
  border-bottom: 0.2rem solid ${GreyBackground};
`;

export const ButtonsWrapper = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  padding: 2rem;
`;

export const Heading = styled(Heading4)`
  font-weight: ${SemiBold};
`;

export const Button = styled(RoundedButton)`
  width: 12rem;
`;

export const Share = styled(I)`
  fill: ${IgnitusBlue};
  margin-right: 0.2rem;
`;

export const ApplyStatus = styled(I)<StatusType>`
  fill: ${White};
  display: ${props => (props.status === 'Applied' ? 'inline-block' : 'none')};
`;
