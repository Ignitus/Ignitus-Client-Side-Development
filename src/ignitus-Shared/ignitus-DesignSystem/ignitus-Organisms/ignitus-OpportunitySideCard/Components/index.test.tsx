import React from 'react';
import { mount, render } from 'enzyme';
import emotionSerializer, { matchers } from 'jest-emotion';
import { InfoCard, Card } from './index';

import { CardType, OpportunityType } from '../types';
import * as S from '../styles';
import { Paragraph } from '../../../ignitus-Atoms/typography';
import { AppIcon } from '../../../../types/iconsTypes/iconEnums';

expect.addSnapshotSerializer(emotionSerializer);
expect.extend(matchers);

const props: CardType = {
  location: 'Australia',
  commitment: 'Full-time',
  salary: '24',
};

const cardProps: OpportunityType = {
  title: 'Location',
  description: 'Australia',
};

it('should rendered', () => {
  const tree = render(<InfoCard {...props} />);
  expect(tree).toMatchSnapshot();
});

describe('<InfoCard />', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = mount(<InfoCard {...props} />);
  });

  it('should render Cards', () => {
    expect(wrapper.find(Card).length).toBe(3);
    expect(
      wrapper
        .find(Card)
        .at(0)
        .prop('title'),
    ).toBe('Location');
    expect(
      wrapper
        .find(Card)
        .at(0)
        .prop('description'),
    ).toBe(props.location);
    expect(
      wrapper
        .find(Card)
        .at(1)
        .prop('title'),
    ).toBe('Commitment');
    expect(
      wrapper
        .find(Card)
        .at(1)
        .prop('description'),
    ).toBe(props.commitment);
    expect(
      wrapper
        .find(Card)
        .at(2)
        .prop('title'),
    ).toBe('Salary');
    expect(
      wrapper
        .find(Card)
        .at(2)
        .prop('description'),
    ).toBe(props.salary);
  });

  it('should render ButtonWrapper', () => {
    expect(wrapper.find(S.ButtonsWrapper).length).toBe(1);
    expect(wrapper.find(S.Button).length).toBe(3);
    expect(
      wrapper
        .find(S.Button)
        .at(0)
        .find('svg').length,
    ).toBe(1);
    expect(
      wrapper
        .find(S.Button)
        .at(0)
        .text(),
    ).toBe('Check IconApply');
    expect(wrapper.find(S.ApplyStatus).length).toBe(1);
    expect(wrapper.find(S.ApplyStatus).prop('name')).toBe(AppIcon.CheckIcon);
    expect(
      wrapper
        .find(S.Button)
        .at(1)
        .text(),
    ).toBe('Save');
    expect(wrapper.find(S.Share).length).toBe(1);
    expect(wrapper.find(S.Share).prop('name')).toBe(AppIcon.ShareIcon);
    expect(
      wrapper
        .find(S.Button)
        .at(2)
        .text(),
    ).toBe('ShareIconShare Link');
  });

  it('should change apply status on Apply button click', () => {
    expect(wrapper.find(S.ApplyStatus).prop('status')).toBe('Apply');
    wrapper
      .find(S.Button)
      .at(0)
      .simulate('click');
    expect(wrapper.find(S.ApplyStatus).prop('status')).toBe('Applied');
  });
});

describe('<Card />', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = mount(<Card {...cardProps} />);
  });
  it('should render Wrapper', () => {
    expect(wrapper.find(S.Wrapper).length).toBe(1);
    expect(wrapper.find(S.Heading).length).toBe(1);
    expect(wrapper.find(S.Heading).text()).toBe(cardProps.title);
    expect(wrapper.find(Paragraph).text()).toBe(cardProps.description);
  });
});
