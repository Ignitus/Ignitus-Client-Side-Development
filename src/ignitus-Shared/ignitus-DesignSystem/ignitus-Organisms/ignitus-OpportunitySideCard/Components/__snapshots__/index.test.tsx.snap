// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should rendered 1`] = `
.emotion-15 {
  border-radius: 1rem;
  background-color: #ffffff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-width: 18rem;
  padding: 2rem;
  border-bottom: 0.2rem solid #F7F7F7;
}

.emotion-0 {
  font-weight: 400;
  font-size: 20px;
  color: #000066;
  font-weight: 600;
}

.emotion-1 {
  font-size: 16px;
  font-weight: 400;
  color: #2D2D2D;
  margin: 0;
}

.emotion-14 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 2rem;
}

.emotion-10 {
  border-radius: 5px;
  border: none;
  cursor: pointer;
  outline: none;
  padding: 16px 32px;
  text-align: center;
  line-height: 1rem;
  margin: 0.5rem;
  outline: none;
  font-size: 18px;
  font-weight: 400;
  -webkit-transition: all 0.3s ease;
  transition: all 0.3s ease;
  outline: none;
  color: #ffffff;
  background-color: #000066;
  padding: 16px 32px;
  font-size: 18px;
  border-radius: 5px;
  border: none;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 30px;
  width: 12rem;
}

.emotion-10:hover,
.emotion-10:focus {
  box-shadow: 0 2px 4px 0 rgba(0,0,0,0.05);
}

.emotion-10 a {
  color: #ffffff;
}

.emotion-10:hover,
.emotion-10:focus {
  background-color: #F9BF64;
  color: #000066;
  outline: none;
}

.emotion-10:hover a,
.emotion-10:focus a {
  color: #000066;
}

.emotion-10:hover svg,
.emotion-10:focus svg {
  fill: #000066;
}

.emotion-9 {
  fill: #ffffff;
  display: none;
}

.emotion-11 {
  border-radius: 5px;
  border: none;
  cursor: pointer;
  outline: none;
  padding: 16px 32px;
  text-align: center;
  line-height: 1rem;
  margin: 0.5rem;
  outline: none;
  font-size: 18px;
  font-weight: 400;
  -webkit-transition: all 0.3s ease;
  transition: all 0.3s ease;
  outline: none;
  color: #000066;
  background-color: #ffffff;
  padding: 16px 32px;
  font-size: 18px;
  border-radius: 5px;
  border: 2px solid #000066;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 30px;
  width: 12rem;
}

.emotion-11:hover,
.emotion-11:focus {
  box-shadow: 0 2px 4px 0 rgba(0,0,0,0.05);
}

.emotion-11 a {
  color: #000066;
}

.emotion-11:hover,
.emotion-11:focus {
  background-color: none;
  color: none;
  outline: none;
}

.emotion-11:hover a,
.emotion-11:focus a {
  color: #000066;
}

.emotion-11:hover svg,
.emotion-11:focus svg {
  fill: #000066;
}

.emotion-13 {
  border-radius: 5px;
  border: none;
  cursor: pointer;
  outline: none;
  padding: 16px 32px;
  text-align: center;
  line-height: 1rem;
  margin: 0.5rem;
  outline: none;
  font-size: 18px;
  font-weight: 400;
  -webkit-transition: all 0.3s ease;
  transition: all 0.3s ease;
  outline: none;
  color: #000066;
  background-color: #E0E0E0;
  padding: 16px 32px;
  font-size: 18px;
  border-radius: 5px;
  border: none;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 30px;
  width: 12rem;
}

.emotion-13:hover,
.emotion-13:focus {
  box-shadow: 0 2px 4px 0 rgba(0,0,0,0.05);
}

.emotion-13 a {
  color: #000066;
}

.emotion-13:hover,
.emotion-13:focus {
  background-color: #C4C4C4;
  color: #000066;
  outline: none;
}

.emotion-13:hover a,
.emotion-13:focus a {
  color: #000066;
}

.emotion-13:hover svg,
.emotion-13:focus svg {
  fill: #000066;
}

.emotion-12 {
  fill: #000066;
  margin-right: 0.2rem;
}

<div
  class="emotion-15"
>
  <div
    class="emotion-2"
  >
    <h4
      class="emotion-0"
    >
      Location
    </h4>
    <br />
    <p
      class="emotion-1"
    >
      Australia
    </p>
  </div>
  <div
    class="emotion-2"
  >
    <h4
      class="emotion-0"
    >
      Commitment
    </h4>
    <br />
    <p
      class="emotion-1"
    >
      Full-time
    </p>
  </div>
  <div
    class="emotion-2"
  >
    <h4
      class="emotion-0"
    >
      Salary
    </h4>
    <br />
    <p
      class="emotion-1"
    >
      24
    </p>
  </div>
  <div
    class="emotion-14"
  >
    <button
      class="emotion-10"
    >
      <svg
        aria-label="Check Icon"
        class="emotion-9"
        fill="none"
        height="17"
        status="Apply"
        viewBox="0 0 23 17"
        width="23"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title>
          Check Icon
        </title>
        <path
          d="M7.291 13.499l-4.52-4.521a1.27 1.27 0 00-1.81 0 1.27 1.27 0 000 1.808l5.413 5.412a1.286 1.286 0 001.821 0L21.887 2.52a1.27 1.27 0 000-1.809 1.27 1.27 0 00-1.809 0L7.291 13.5z"
        />
      </svg>
      Apply
    </button>
    <button
      class="emotion-11"
    >
      Save
    </button>
    <button
      class="emotion-13"
    >
      <svg
        aria-label="ShareIcon"
        class="emotion-12"
        fill="none"
        height="20"
        viewBox="0 0 22 20"
        width="22"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title>
          ShareIcon
        </title>
        <path
          d="M5.5 15C7.58 5.119 15.583 3.333 15.583 3.333V0L22 6.637l-6.417 6.696V10S9.928 9.829 5.5 15zm11 .145V18H1.833V6h6.048a16.033 16.033 0 012.145-2H0v16h18.333v-6.769L16.5 15.145z"
          fill="#006"
        />
      </svg>
      Share Link
    </button>
  </div>
</div>
`;
